src/libcore/any.rs

```diff
@@ -446,14 +446,16 @@ impl TypeId {
 /// # Note
 ///
 /// This is intended for diagnostic use. The exact contents and format of the
-/// string are not specified, other than being a best-effort description of the
-/// type. For example, `type_name::<Option<String>>()` could return the
-/// `"Option<String>"` or `"std::option::Option<std::string::String>"`, but not
-/// `"foobar"`. In addition, the output may change between versions of the
-/// compiler.
+/// string returned are not specified, other than being a best-effort
+/// description of the type. For example, amongst the strings
+/// that `type_name::<Option<String>>()` might map to are `"Option<String>"` and
+/// `"std::option::Option<std::string::String>"`.
```

Comment:
I think we could keep mentioning "not `"foobar"`". And using "map to" as a return value sounds strange. I prefer keeping the previous description.

> For example, `type_name::<Option<String>>()` could return `"Option<String>"` or `"std::option::Option<std::string::String>"`, but will not return irrelevant description like `"foobar"`.

Reply:
Fair enough about "map to" -- fixed in https://github.com/rust-lang/rust/pull/71870/commits/003ed802c4a289d959d4212ad80f33df7f3e2bc3.

Personally, I found the "irrelevant description" confusing, because it suggests a level of guarantee about the output that `type_name` doesn't really provide. [This whole PR came about when I was surprised at some of the things it returns.] So I was trying to make clear that `type_name::<Option<String>>()` might return who-knows-what, and that the two strings mentioned certainly aren't the only plausible ones (e.g. `::std::option::Option<::std::string::String>` might be another similar string), with the "amongst the things it can return are ..." formulation. I'm happy to be corrected, and/or to reconsider alternative phrasings though!